napari/_vispy/_tests/test_vispy_image_layer.py

```diff
@@ -0,0 +1,84 @@
+from itertools import permutations
+from typing import Union
+
+import numpy as np
+import pytest
+from vispy.visuals import ImageVisual, VolumeVisual
+from vispy.visuals.transforms.linear import STTransform
+
+from napari._vispy.layers.image import VispyImageLayer
+from napari.layers import Image
+
+
+def _node_scene_size(node: Union[ImageVisual, VolumeVisual]) -> np.ndarray:
+    """Calculates the size of a vispy node in 3D scene homogeneous coordinates."""
```

Comment:
Just to clarify: this means that the return value should be a 4-tuple `(x_size, y_size, z_size, 1)` where `xyz` are canvas coordinates (with `z` out of the screen)?

Reply:
Yes, though I'm not sure I'd refer to them as canvas coordinates as I think of the canvas as strictly 2D. I think my use of scene is also not quite the same as vispy's (though it's close!).

Reply:
I modified/expanded the docstring and made this function return a 3-vector since the last coordinate should always be 1 and we don't care about it.